standalone/src/test/java/io/pravega/local/AuthEnabledInProcPravegaClusterTest.java

```diff
@@ -52,105 +38,61 @@ public void setUp() throws Exception {
+    }
 
-        String readerGroup = UUID.randomUUID().toString().replace("-", "");
-        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
-                .stream(Stream.of(scope, streamName))
-                .disableAutomaticCheckpoints()
+    @Override
+    ClientConfig prepareValidClientConfig() {
+        return ClientConfig.builder()
+                .controllerURI(URI.create(localPravega.getInProcPravegaCluster().getControllerURI()))
+                .credentials(new DefaultCredentials("1111_aaaa", "admin"))
                 .build();
-
-        @Cleanup
-        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, clientConfig);
-        readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
-
-        @Cleanup
-        EventStreamReader<String> reader = clientFactory.createReader(
-                "readerId", readerGroup,
-                new JavaSerializer<String>(), ReaderConfig.builder().build());
-
-        // Keeping the read timeout large so that there is ample time for reading the event even in
-        // case of abnormal delays in test environments.
-        String readEvent1 = reader.readNextEvent(10000).getEvent();
-        log.info("Done reading event [{}]", readEvent1);
-
-        String readEvent2 = reader.readNextEvent(10000).getEvent();
-        log.info("Done reading event [{}]", readEvent2);
-
-        assertEquals(writeEvent1, readEvent1);
-        assertEquals(writeEvent2, readEvent2);
     }
 
-    @Test
-    public void testCreateStreamFailsWhenCredentialsAreInvalid() {
-        Assert.assertNotNull("Pravega not initialized", localPravega);
-        String scope = "Scope";
-
-        ClientConfig clientConfig = ClientConfig.builder()
+    /**
+     * This test verifies that create stream fails when the client config is invalid.
+     *
+     * Note: The timeout being used for the test is kept rather large so that there is ample time for the expected
+     * exception to be raised even in case of abnormal delays in test environments.
+     */
+    @Test(timeout = 300000)
+    public void testCreateStreamFailsWithInvalidClientConfig() {
+       ClientConfig clientConfig = ClientConfig.builder()
                 .credentials(new DefaultCredentials("", ""))
                 .controllerURI(URI.create(localPravega.getInProcPravegaCluster().getControllerURI()))
                 .build();
 
         @Cleanup
         StreamManager streamManager = StreamManager.create(clientConfig);
 
-        AssertExtensions.assertThrows(RuntimeException.class,
-                () -> streamManager.createScope(scope));
+        try {
+            streamManager.createScope(scopeName());
+        } catch (RetriesExhaustedException e) {
+            if (!hasAuthExceptionAsRootCause(e)) {
+                throw e;
+            }
+        }
+    }
+
+    private boolean hasAuthExceptionAsRootCause(Throwable e) {
+        Throwable innermostException = ExceptionUtils.getRootCause(e);
```

Comment:
Please use our utility class called Exceptions.unwrap

Reply:
The `Exceptions.unwrap` won't work here, as it only unwraps `CompletionException`  or `ExecutionException`, but here we have a `RetriesExhaustedException`.

Reply:
I suppose we could add `RetriesExhaustedException` to `Exceptions.unwrap`. I don't see an issue with that.

@shiveshr and @tkaitchuck: do you anticipate any hidden bugs if we do this? There is no value to a `RetriesExhaustedException` **as long as** it contains a meaningful inner exception (if it doesn't, then we shouldn't unwrap)

Reply:
I tried adding `RetriesExhaustedException` to the whitelist used in Exceptions.unwrap(), but there was an an unintended semantical side-effect: the `StreamManager.createScope(...)` method then returns `StatusRuntimeException` as opposed to `RetriesExhaustedException` as it does today. There might be other side-effects in other paths.

All I need here is to fetch the innermost/root exception and check if it is related to auth. The `org.apache.commons.lang3.exception.ExceptionUtils.getRootCause(e)` used here does just that. That being said, if we do need to abstract out the unwrapping of the root exception through our own `Exceptions` utility, one way to do that is to write a new method in the `Exceptions` class that delegates the call to `ExceptionUtils.getRootCause()`; I don't see much benefit of that though unless there is a logic that we needed to add on top of that.  

What do you guys say?

Reply:
My suggestion is we separate the two in different issues -
for the sake of this test, we use Exceptions.unwrap + RetriesExhausted 

And have a separate issue dealing with RetriesExhausted as part of Exceptions.unwrap where we can evaluate if it causes any regressions or is safe to do so.

Reply:
Please take a look at the code in the latest commit. I still use the ExceptionUtils for reasons mentioned in my last comment in the thread. This should cover the intent of the first paragraph of @shiveshr's comments. Please unresolve this conversation if you are not satisfied.